update-help-messages.sh

```diff
@@ -25,37 +33,45 @@ function print_help() {
     help_text="$(echo "$help_text" | sed 's/^/   /')"
 
     # Print the command name in a rst code block
-    echo "::"
-    echo
-    echo "$help_text"
-    echo
+    {
+        echo "::"
+        echo
+        echo "$help_text"
+        echo
+    } >"$file_path"
 }
 
-# Loop through each command and store help output in separate files
-# `pgcopydb --help` output will be in `pgcopydb.rst`. All other commands will
-# be in their own file, e.g. `pgcopydb clone --help` will be in `clone.rst`
-function print_help_to_files() {
-    declare -a cmds=( \
-        "clone" "follow" "copy" "snapshot" "compare" "dump" "restore" "list" "stream" "ping" \
-        "compare schema" "compare data" \
-        "copy db" "copy roles" "copy extensions" "copy schema" "copy data" "copy table-data" "copy blobs" "copy sequences" "copy indexes" "copy constraints" \
-        "dump schema" "dump pre-data" "dump post-data" "dump roles" \
-        "list databases" "list extensions" "list collations" "list tables" "list table-parts" "list sequences" "list indexes" "list depends" "list schema" "list progress" \
-        "restore schema" "restore pre-data" "restore post-data" "restore roles" "restore parse-list" \
-        "stream setup" "stream cleanup" "stream prefetch" "stream catchup" "stream replay" "stream sentinel" "stream receive" "stream transform" "stream apply" \
-        "stream sentinel create" "stream sentinel drop" "stream sentinel get" "stream sentinel set" \
-        "stream sentinel set startpos" "stream sentinel set endpos" "stream sentinel set apply" "stream sentinel set prefetch" \
-    )
-
-    # Print the main pgcopydb help output to pgcopydb.rst
-    print_help "" >"$INCLUDE_DIR/pgcopydb.rst"
-
-    # Print the help output for each command to their separate file
-    for cmd in "${cmds[@]}"; do
-        print_help "$cmd" >"$INCLUDE_DIR/$cmd.rst"
-    done
+# Parse the output of `pgcopydb help` and call print_help_to_file for each command
+function parse_help_output() {
+
+    # Loop over all the lines of the help text, parse commands and subcommands,
+    # and call print_help_to_file for each command.
+    while read -r l; do
+        # Parse first section of the help text
+        if [[ $l =~ ^pgcopydb$ ]]; then
+            cmd=""
+            subcmd=""
+        # Parse other sections of the help text that contain `pgcopydb <cmd>`
+        elif [[ $l =~ ^pgcopydb\ (.+) ]]; then
+            cmd="${BASH_REMATCH[1]}"
+            subcmd=""
+            continue;
+        # Parse subcommands
+        elif [[ $l =~ ^([a-z-]+) || $l =~ ^\+\ ([a-z-]+) ]]; then
```

Comment:
I think these regexps needs comments to explain what you expect from them, and why you had to separate them in two rounds of processing instead of one. Usually I would put in the main comment for the first `if` a list of examples of lines that are parsed and what we get from the regexps etc, in a pretty visual way.

Again, it's just a shell script, but we have to maintain it as soon as it's merged.